### ISSUE TITLE
Feature Enhancement: Improve completion detection and shutdown to support no embedded applications

### ISSUE THREAD

**Issue by greghaynes** [cnoe-io/idpbuilder#76]
Today, idpbuilder shuts down by cancelling it's context whenever it completes reconciling the argo applications (https://github.com/cnoe-io/idpbuilder/blob/main/pkg/controllers/localbuild/controller.go#L254). This means that if this controller is run in a mode where embedded argo applications are disabled, it causes the entire controller manager to shutdown without any applications being installed.

Consider a case where this project is being used as an integration testing framework and therefore we would like to disable all the embedded applications and install them ourselves out of band. If we share a controller manager with the localbuild reconciler this is impossible as it will shut down the controller manager.

**Comment by nabuskey** [comment]
Agreed. We need work on this. Another case I can think of is: If we want to allow for continuously syncing local files to gitea repositories, the controller manager should not shut down prematurely.
